editor/plugins/parallax_background_editor_plugin.cpp

```diff
@@ -0,0 +1,145 @@
+
+void ParallaxBackgroundEditorPlugin::edit(Object *p_object) {
+	parallax_background = Object::cast_to<ParallaxBackground>(p_object);
+}
+
+bool ParallaxBackgroundEditorPlugin::handles(Object *p_object) const {
+	return p_object->is_class("ParallaxBackground");
+}
+
+void ParallaxBackgroundEditorPlugin::make_visible(bool p_visible) {
+	if (p_visible) {
+		toolbar->show();
+	} else {
+		toolbar->hide();
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::_menu_callback(int p_idx) {
+	switch (p_idx) {
+		case MENU_CONVERT_TO_PARALLAX_2D: {
+			convert_to_parallax2d();
+		} break;
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::convert_to_parallax2d() {
+	ParallaxBackground *parallax_bg = parallax_background;
+	TypedArray<Node> children = parallax_bg->get_children();
+
+	EditorUndoRedoManager *ur = EditorUndoRedoManager::get_singleton();
+	ur->create_action(TTR("Convert to Parallax2D"));
+
+	for (int i = 0; i < children.size(); i++) {
+		ParallaxLayer *parallax_layer = Object::cast_to<ParallaxLayer>(children[i]);
+
+		if (!parallax_layer) {
+			continue;
+		}
+
+		Parallax2D *parallax2d = memnew(Parallax2D);
+
+		Point2 offset = parallax_bg->get_scroll_base_offset() * parallax_layer->get_motion_scale();
+		parallax2d->set_scroll_offset(offset + parallax_layer->get_position());
+
+		Point2 limit_begin = parallax2d->get_limit_begin();
+		Point2 limit_end = parallax2d->get_limit_end();
+
+		if (parallax_bg->get_limit_begin().x != 0 || parallax_bg->get_limit_end().x != 0) {
+			limit_begin.x = parallax_bg->get_limit_begin().x;
+			limit_end.x = parallax_bg->get_limit_end().x;
+		}
+
+		if (parallax_bg->get_limit_begin().y != 0 || parallax_bg->get_limit_end().y != 0) {
+			limit_begin.y = parallax_bg->get_limit_begin().y;
+			limit_end.y = parallax_bg->get_limit_end().y;
+		}
+
+		parallax2d->set_limit_begin(limit_begin);
+		parallax2d->set_limit_end(limit_end);
+		parallax2d->set_follow_viewport(!parallax_bg->is_ignore_camera_zoom());
+		parallax2d->set_repeat_size(parallax_layer->get_mirroring());
+		parallax2d->set_scroll_scale(parallax_layer->get_motion_scale());
+
+		SceneTreeDock::get_singleton()->replace_node(parallax_layer, parallax2d);
+	}
+
+	if (parallax_bg->is_ignore_camera_zoom()) {
+		CanvasLayer *canvasLayer = memnew(CanvasLayer);
+		SceneTreeDock::get_singleton()->replace_node(parallax_bg, canvasLayer);
+	} else {
+		Node2D *node2d = memnew(Node2D);
+		SceneTreeDock::get_singleton()->replace_node(parallax_bg, node2d);
+	}
+
+	ur->commit_action(false);
+}
+
+void ParallaxBackgroundEditorPlugin::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			menu->get_popup()->connect("id_pressed", callable_mp(this, &ParallaxBackgroundEditorPlugin::_menu_callback));
+			menu->set_icon(menu->get_editor_theme_icon(SNAME("ParallaxBackground")));
+		} break;
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::_bind_methods() {
+}
+
+ParallaxBackgroundEditorPlugin::ParallaxBackgroundEditorPlugin() {
+	parallax_background = nullptr;
```

Comment:
```suggestion
```
Already initialized in the header.

Reply:
I thought this was odd, but `cpu_particles_2d_editor_plugin.cpp` does this too, so I assumed it was for a good reason.

Reply:
It's not. It's probably a leftover from when the properties were initialized in constructors. We moved them to headers around Godot 4.0.